src/SamplesApp/SamplesApp.Shared/App.xaml.cs

```diff
@@ -129,11 +129,53 @@ override void OnLaunched(LaunchActivatedEventArgs e)
 			DisplayLaunchArguments(e);
 		}
 
+		protected
+#if HAS_UNO
+			internal
+#endif
+			override async void OnActivated(IActivatedEventArgs e)
+		{
+			base.OnActivated(e);
+
+			Frame rootFrame = Windows.UI.Xaml.Window.Current.Content as Frame;
+
+			if (rootFrame == null)
+			{ 
+				// Create a Frame to act as the navigation context and navigate to the first page
+				rootFrame = new Frame();
+
+				rootFrame.NavigationFailed += OnNavigationFailed;
+
+				// Place the frame in the current Window
+				Windows.UI.Xaml.Window.Current.Content = rootFrame;
+			}
+
+			if (rootFrame.Content == null)
+			{
+				// When the navigation stack isn't restored navigate to the first page,
+				// configuring the new page by passing required information as a navigation
+				// parameter
+				rootFrame.Navigate(typeof(MainPage));
+			}
+
+			Windows.UI.Xaml.Window.Current.Activate();
```

Comment:
We probably should not duplicate this part, and take it from the OnLaunched method.

Reply:
Wanted to do that but forgot to get back to it 😀